addon/-private/system/relationships/state/relationship.js

```diff
@@ -77,6 +77,7 @@ export default class Relationship {
     this.meta = null;
     this.hasData = false;
     this.hasLoaded = false;
+    this.hasLocalData = false;
```

Comment:
What's the case where `hasData && !hasLocalData`?

ie how does "local" data differ from data?

Does this capture explicit `data` in a json-api resource as opposed to data loaded from a link in a json-api resource that initially contained no `data`?

Reply:
Maybe (probably) because I'm french, but I would (wrongly?) better understand `relationshipHasDataProperty`.